### ISSUE TITLE
ZstdCompressor(threads=-1).compress(...) does not use multithreaded compression

### ISSUE THREAD

**Issue by thedrow** [indygreg/python-zstandard#27]
When I run:
```
cctx = zstd.ZstdCompressor(threads=-1)
compressed = cctx.compress(b'data to compress')
```
8 threads are initialized as expected but single threaded compression is used nevertheless.

It looks like the compression parameters are not being used  https://github.com/indygreg/python-zstandard/blob/master/zstd/compress/zstd_compress.c#L2930

**Comment by indygreg** [comment]
This is by design. Sort of. `compress()` uses the simple, one-shot compress API. That zstandard C API doesn't support multi-threaded compression.

Even if it did, it doesn't make that much sense. To benefit from multi-threaded compression, you need lots of data. And the one-shot `compress()` API doesn't return until compression has finished. For that amount of data, a streaming API is preferred. That being said, if the C API exposed a multi-threaded one-shot API, python-zstandard would likely support it. Because flexibility is good.

**Comment by thedrow** [comment]
In that case, initialize the threads lazily and raise an error when a user initializes ZstdCompressor with threads and tries to use the simple one shot compress API.
Otherwise, people will waste resources and time to investigate this problem like we did.

**Comment by terrelln** [comment]
The zstd-1.3.0 has a new advanced API which can support 1 shot multithreaded compression.

```c
/* Error checking omitted */
ZSTD_inBuffer input = /* */;
ZSTD_outBuffer output = /* */;
ZSTD_CCtx* const zc = ZSTD_createCCtx();
ZSTD_CCtx_setParameter(zc, ZSTD_p_nbThreads, 32);
ZSTD_CCtx_setParameter(zc, ZSTD_p_compressionLevel, 19);
ZSTD_compress_generic(zc, output, input, ZSTD_e_end);
ZSTD_freeCCtx();
```

**Comment by indygreg** [comment]
I've ported `python-zstandard` to the new advanced compression API available in zstd 1.3. This should work on current `master`. It will be part of the `0.9` release.
